lib/dl_core/dl_core/connection_executors/common_base.py

```diff
@@ -58,6 +58,17 @@ class ExecutionMode(enum.Enum):
     RQE = enum.auto()
 
 
+class RQEMode(enum.Enum):
+    INTERNAL = enum.auto()
+    EXTERNAL = enum.auto()
+
+
+@attr.s
+class ExecutionSettings:
+    execution_mode: ExecutionMode = attr.ib()
+    rqe_mode: RQEMode | None = attr.ib(default=None)
```

Comment:
i would merge it in single enum:

direct, internal_rqe, external_rqe

to avoid impossible combination such as
execution_mode=DIRECT
rqe_mode=None